google/ggrc-core
# GGRC-2715 Migrate program roles to ACL

# TODO

- [x] Migration (might need some cleanup)
- [x] Update the frontend (Program roles order isn't ok)
- [x] Update csv files and corresponding tests
- [x] Fix failing tests (selenium tests might need a few updates)
- [x] Propagation event handlers
- [x] Audit Archiving fixes 
- [x] ACL Update to prevent Program Editors from adding ACL records

# Issue description

Migrates Programs to ACL.

Program role propagation:

```
Program
- Mapped objects (through relationships)
  - Comments & Documents
  - Proposals & Revisions
- Audit
  - Assessment Templates
  - Snapshots
  - Assessments
    - Comments & Documents
  - Issues
    - Comments & Documents
```

# Steps to test the changes

Run the migration, make sure the above propagation scheme works in all possible ways (through imports, through object creation, object deletion etc)

# Solution description

Same as Audit ACL.

# Sanity checklist



# Migration checklist


# PR Review checklist
